mtp_prisoner_location_admin/apps/prisoner_location_admin/tests/test_forms.py

```diff
@@ -0,0 +1,43 @@
+from django.core.urlresolvers import reverse
+from django.test import SimpleTestCase, RequestFactory
+
+from ..forms import LocationFileUploadForm
+from . import get_csv_data_as_file, generate_testable_location_data
+
+
+class LocationFileUploadFormTestCase(SimpleTestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_location_file_valid(self):
+        file_data, _ = generate_testable_location_data()
+
+        request = self.factory.post(
+            reverse('location_file_upload'),
+            {'location_file': get_csv_data_as_file(file_data)}
+        )
+        form = LocationFileUploadForm(request.POST, request.FILES, request=request)
+
+        self.assertTrue(form.is_valid())
+
+    def test_location_file_short_row_length_invalid(self):
+        file_data, _ = generate_testable_location_data()
+        file_data += '\nA1234GY,1997-9-2'
+
+        request = self.factory.post(
+            reverse('location_file_upload'),
+            {'location_file': get_csv_data_as_file(file_data)}
+        )
+        form = LocationFileUploadForm(request.POST, request.FILES, request=request)
+
+        self.assertFalse(form.is_valid())
```

Comment:
I'd check the error msg as well, just to be sure that it's that particular validation error being raised and not a different one.

If you don't like copying/pasting the string here maybe you can extend the error_messages on the form field and reference it here.